gef.py

```diff
@@ -10347,23 +10366,25 @@ def maps(self) -> List[Section]:
 
     def __parse_maps(self) -> List[Section]:
         """Return the mapped memory sections"""
-        try:
-            if is_qemu_system():
-                return list(self.__parse_info_mem())
-        except gdb.error:
-            # Target may not support this command
-            pass
-        try:
-            return list(self.__parse_procfs_maps())
-        except FileNotFoundError:
-            return list(self.__parse_gdb_info_sections())
+        if gef.arch.maps is not None:
+            # print("Trying to call architecture's map provider")
+            maps = list(gef.arch.maps())
+            return maps
+
+        return list(itertools.chain(self.parse_monitor_info_mem(),
+                                    self.parse_procfs_maps(),
+                                    self.parse_gdb_info_sections()))
 
-    def __parse_procfs_maps(self) -> Generator[Section, None, None]:
+    # TODO: GefMemoryMapProvider
+    # TODO: Might be easier to just return nothing on error so we can 'chain' to the next
+    @staticmethod
+    def parse_procfs_maps() -> Generator[Section, None, None]:
         """Get the memory mapping from procfs."""
         procfs_mapfile = gef.session.maps
         if not procfs_mapfile:
             is_remote = gef.session.remote is not None
-            raise FileNotFoundError(f"Missing {'remote ' if is_remote else ''}procfs map file")
+            # raise FileNotFoundError(f"Missing {'remote ' if is_remote else ''}procfs map file")
```

Comment:
removed for now since i am using itertools chain. Have to decide whether an error just returns nothing (to try the next) or if we just re-add the try/except logic from before.